doc/design/networking/Investigation-on-NSX-integration.md

```diff
@@ -0,0 +1,254 @@
+      * NSX Objects: Logical Switches, Security Tags, Logical Routers, etc. 
+    In this document, we propose to use vNICs as the grouping criteria. We will discuss more on this later.
+
+  * Security Policies
+
+    NSX provides security policies as a way to group rules for security controls that will be applied to one or more security groups.
+    Each security policy contains the following :
+      * Firewall rules: 
+        * NSX built in distributed firewall.
+      * weight
+        * Weight of a policy determines the rank of the policy versus other policies in the NSX eco-system. Higher weight rules are executed first.
+
+## An Overview of Workflow
+
+Current VIC implementation may have the container bridge networks all reside on a single port group and only be separated by IP space. Our container networks (created using `docker network create`) are not isolated from each other. For example, a containerVM in one container network can change its IP and be able to reach containerVMs in another container network.
+
+Our research shows that we can use NSX security groups and policies on top of current VIC implementation and achieve micro-segmentation. 
+
+Our goals :
+  * VMs can communicate with VMs in the same network (security group).
+  * VMs cannot talk to VMs in a differrent network (security group).
+  * one VM to be added to more than one network (security group).
+
+In the following example, we see that : 
+  * containerVM1 and containerVM2 are in default bridge network
+  * containerVM3 , containerVM4 and containerVM5 are in bridge network `network-a` 
+  * containerVM5 and containerVM6 are in bridge network `network-b`
+    (note that adding containerVM5 to both network `network-a` and network `network-b` are not supported on current VIC yet.)
+
+![An Example of How We Apply Security Group on Container Networks](pics/example-SG-vs-network.png)
+
+Let us use this example to explain the workflow.
+
+### Prerequisite
+We assume vSphere admin already has NSX installed and configured on vCenter. That means there is an NSX logical switch created by the admin, which automatically creates a port group on distributed switch.
+So this port group can be used as a bridge network for `vic-machine create`
+
+![Distributed Switch with logic switch added](pics/dswitch.png)
+
+### Create VCH 
+Specifying 'bridge-network' as the port group 'vxw-dvs-55-virtualwire-1-sid-5000-logical-switch-1'.
+  * VIC creates VCH
+  * NSX manager creates a security group `<vch name>-SG-VCH` with dynamic membership: NIC (of VCH) 
+  * NSX manager creates these security policies shown in the following picture
+  * NSX manager applies security policy `<vch name>-SP-0` on security group `<vch name>-SG-VCH`
+
+  ![Security policies](pics/security-policies.png)
+
+### docker run --name=containerVM1 <image name>
+  * VIC create containerVM1
+  * NSX manager creates security group `<vch name>-SG-bridge` with dynamic membership: NIC-1 (of containerVM1)
```

Comment:
Sorry for the confusion here. I mean we create a security group for each container network. And then everytime we add a container (say with NIC-1) to that network, we update the security group by adding a criteria of 'entitiy equals to NIC-1'. This is how the container's NIC-1 to the security group.